source/common/network/utility.cc

```diff
@@ -156,6 +156,53 @@ Address::InstanceConstSharedPtr Utility::parseInternetAddress(const std::string&
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+Address::InstanceConstSharedPtr
+Utility::parseInternetAddressAndPortNoThrow(const std::string& ip_address, bool v6only) {
+  if (ip_address.empty()) {
+    return nullptr;
+  }
+  if (ip_address[0] == '[') {
+    // Appears to be an IPv6 address. Find the "]:" that separates the address from the port.
+    const auto pos = ip_address.rfind("]:");
+    if (pos == std::string::npos) {
+      return nullptr;
+    }
+    const auto ip_str = ip_address.substr(1, pos - 1);
+    const auto port_str = ip_address.substr(pos + 2);
+    uint64_t port64 = 0;
+    if (port_str.empty() || !absl::SimpleAtoi(port_str, &port64) || port64 > 65535) {
+      return nullptr;
+    }
+    sockaddr_in6 sa6;
+    memset(&sa6, 0, sizeof(sa6));
+    if (ip_str.empty() || inet_pton(AF_INET6, ip_str.c_str(), &sa6.sin6_addr) != 1) {
+      return nullptr;
+    }
+    sa6.sin6_family = AF_INET6;
+    sa6.sin6_port = htons(port64);
+    return std::make_shared<Address::Ipv6Instance>(sa6, v6only);
+  }
+  // Treat it as an IPv4 address followed by a port.
+  const auto pos = ip_address.rfind(':');
+  if (pos == std::string::npos) {
+    return nullptr;
+  }
+  const auto ip_str = ip_address.substr(0, pos);
+  const auto port_str = ip_address.substr(pos + 1);
+  uint64_t port64 = 0;
+  if (port_str.empty() || !absl::SimpleAtoi(port_str, &port64) || port64 > 65535) {
+    return nullptr;
+  }
+  sockaddr_in sa4;
+  memset(&sa4, 0, sizeof(sa4));
+  if (ip_str.empty() || inet_pton(AF_INET, ip_str.c_str(), &sa4.sin_addr) != 1) {
+    return nullptr;
+  }
+  sa4.sin_family = AF_INET;
+  sa4.sin_port = htons(port64);
+  return std::make_shared<Address::Ipv4Instance>(&sa4);
+}
+
 Address::InstanceConstSharedPtr Utility::parseInternetAddressAndPort(const std::string& ip_address,
                                                                      bool v6only) {
   if (ip_address.empty()) {
```

Comment:
I think the idea is that the `parseInternetAddressAndPort()` calls `parseInternetAddressAndPortNoThrow()` and if it returns nullptr it then calls `throwWithMalformedIp(ip_address);` This will reduce code duplication.

Reply:
Looks like I didn't push all of my local commits to remote, let me check.

Reply:
Yep, didn't push all changes. Here we go https://github.com/envoyproxy/envoy/pull/14015/commits/f25849690f4a8def557ebdec6e5e929fa03b8b33